src/theme/base/_spacing.scss

```diff
@@ -0,0 +1,104 @@
+/**
+ * We need a way to consistently set the spacing in the app. This script
+ * generates classnames for paddings and margins. It also defines a mixin to
+ * use inside responsive selectors.
+ *
+ * The idea is to use composition when possible, for responsive scenarios we should use
+ * the `spacing` mixing.
+ *
+ * The classnames consist of three parts, representing the following:
+ *
+ *  .[Type][Direction][Size]
+ *
+ *  Type = `M` or `P` = Margin or Padding
+ *  Direction = `T`, `L`, `R`, `B` = Top, Left, Right, Bottom
+ *  Size = `Z`, `T`, `S`, `M`, `L`, `XL` = Zero, Tiny, Small, Medium, Large, Extra Large
+ *
+ **/
+$typeList: (
+  'M': 'margin',
+  'P': 'padding',
+);
+
+$directionListVertical: (
+  'T': ('top'),
+  'B': ('bottom'),
+  'V': ('top', 'bottom'),
+);
+
+$directionListHorizontal: (
+  'L': ('left'),
+  'R': ('right'),
+  'H': ('left', 'right'),
+);
+
+// Size list description:
+// Z - zero
+// T - tiny
+// E - extra small
+// S - small
+// M - medium
+// L - large
+// XL - Extra large
+$sizeList: (
```

Comment:
I think is better to just have the long name since it's more usable and gives more information
```
auto,
tinny,
left,
right,
...
```

Reply:
It can be intimidating at first, but you get used to pretty soon! In two weeks you will be really comfortable, I can assure you that!

We could use the actual name for classes if that makes things easier: `MarginTopSmall` or `PaddingBottomXLarge`, the only thing is that our css bundle will increase, maybe not a lot but something to consider.

Have you ever used tailwind (https://tailwindcss.com/)? This is a similar approach.

Reply:
I've never used it! I don't know how much good it gives, just saying, we can try. But, I am not hundred per cent convinced 😛